piecrust-uplink/src/lib.rs

```diff
@@ -4,13 +4,69 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
+//! Piecrust Uplink is the crate for writing WASM smart-contracts in Rust
```

Comment:
Hyphenation of `smart-contracts` is fine here, given it's a compound adjective in this context, but the convention we seem to have used elsewhere is `smart contracts`. So I'd recommend either using `smart-contract` here or changing `smart contract` in other places to be hyphenated to remain consistent.